README.md

```diff
@@ -0,0 +1,57 @@
+
+## Motivation
+
+Development productivity of large scientific codes, like geophysical fluid dynamics (GFD) models, decreased drastically in recent times due to the fact these community models often have to run efficiently on multiple computing architectures that impose the use of different programming models. Embedded domain specific languages (EDSLs) in C ++ have been used successfully in the past to enable a clear separation of concerns between the domain algorithms and the implementation strategy, thus allowing a single source code which is performance portable across heterogeneous architectures. Yet, using EDSLs in a correct and efficient manner often requires expert knowledge in high-performance computing. In addition, current DSLs are usually designed and developed for a specific scientific model with little to no reusability among DSLs.
+
+We introduce a new compiler framework, consisting of GTclang and Dawn, that decouples optimization and code generation from high level DSLs. By exposing a Stencil Intermediate Representation (SIR), we allow users to share the toolchain to optimize and generate code among several DSLs. This allows the design of expressive, concise DSLs that can focus on applicability and don't need to bother with the aspect of high-performance computing. Using Dawn as a common compiler infrastructure can drastically reduce development and maintenance effort, while increasing the performance of the generated code, for new and existing DSLs in the GFD model domain.
+
+## Core Features
+
+* GTClang translates an easy to understand but expressive DSL that is capable of modelling Finite Difference stencils as well as spare solvers into a relatively simple SIR. See the `README` in the GTClang subdirectory for an illustrative example
+* Dawn allows the user to generate fast performing code for several back-ends from the SIR.
+* Dawn exposes several APIs in different languages (C++, Java, Python) to parse and process the SIR. 
+* Dawn is able to generate code to be run on Distributed Memory Machines based on MPI, Machines with access to GPUs based on CUDA as well as naive C++ code with close to no parallelism for debugging.
+* Dawn offers a wide range of optimization and static analysis passes to guarantee correctness as well as performance of the generated parallel program.
+
+## Building
+
+Even though **GTClang** and **dawn** share a common repository, they are built independently from each other. Dawn has to be built first, GTClang is then built "on top" of dawn in a second step. Please see the [dawn README](https://github.com/MeteoSwiss-APN/dawn/blob/master/dawn/README.md) and [GTClang README](https://github.com/MeteoSwiss-APN/dawn/blob/master/dawn/README.md) for detailed build and testing instructions. Dependencies include: 
+
+* C++-17 compatible compiler
+* clang > 6.0 and containing llvm
```

Comment:
gtclang isn't building for me with clang 9. Either we need to correct this or change the phrasing here.

Reply:
Can you post the problem with clang 9 as an issue. I have a branch where I try to get it working with llvm master. I am curious which of these fixes is needed for clang 9.

Reply:
Should be resolved by #327.